samples/sensor/wsen_itds/src/main.c

```diff
@@ -0,0 +1,164 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+
+#include <sys/printk.h>
+#include <sys_clock.h>
+#include <stdio.h>
+
+#include <device.h>
+#include <drivers/sensor.h>
+#include <drivers/i2c.h>
+
+#define MAX_TEST_TIME	5000
+#define SLEEPTIME	300
+
+static void print_accl_data(struct device *itds)
+{
+	struct sensor_value val[3];
+
+	if (sensor_channel_get(itds, SENSOR_CHAN_ACCEL_XYZ, val) < 0) {
+		printf("Cannot read itds accl channels.\n");
+		return;
+	}
+
+	printf("Accl (m/s): X=%f, Y=%f, Z=%f\n",
+	       val[0].val1 + val[0].val2 / 1000000.0,
+	       val[1].val1 + val[1].val2 / 1000000.0,
+	       val[2].val1 + val[2].val2 / 1000000.0);
+}
+
+static void print_temp_data(struct device *itds)
+{
+	struct sensor_value val;
+
+	if (sensor_channel_get(itds, SENSOR_CHAN_DIE_TEMP, &val) < 0) {
+		printf("Temperature channel read error.\n");
+		return;
+	}
+
+	printf("Temperature (Celsius): %f\n",
+	       val.val1 + val.val2 / 1000000.0);
+}
+
+static void test_polling_mode(struct device *itds)
+{
+	int32_t remaining_test_time = MAX_TEST_TIME;
+
+	do {
+		if (sensor_sample_fetch(itds) < 0) {
+			printf("accl sample update error.\n");
+		}
+
+		print_accl_data(itds);
+
+		print_temp_data(itds);
+
+		/* wait a while */
+		k_sleep(K_MSEC(SLEEPTIME));
+
+		remaining_test_time -= SLEEPTIME;
+	} while (remaining_test_time > 0);
+}
+
+#if defined(CONFIG_ITDS_TRIGGER)
+static void trigger_handler(struct device *itds,
+			    struct sensor_trigger *trigger)
+{
+	switch (trigger->type) {
+	case SENSOR_TRIG_DATA_READY:
+		printf("any drdy.\n");
+		if (sensor_sample_fetch(itds) < 0) {
+			printf("sample update error.\n");
+		}
+
+		print_accl_data(itds);
+		return;
+
+	default:
+		printf("trigger handler: unknown trigger type.\n");
+		return;
+	}
+}
+#endif
+
+static void test_trigger_mode(struct device *itds)
+{
+#if defined(CONFIG_ITDS_TRIGGER)
+	s32_t remaining_test_time = MAX_TEST_TIME;
+	struct sensor_trigger trig;
+	struct sensor_value attr;
+
+	printf("Testing data ready trigger.\n");
+
+	attr.val1 = 400;
+	attr.val2 = 0;
+
+	if (sensor_attr_set(itds, SENSOR_CHAN_ACCEL_XYZ,
+			    SENSOR_ATTR_SAMPLING_FREQUENCY, &attr) < 0) {
+		printf("cannot set sampling frequency.\n");
+		return;
+	}
+
+	trig.type = SENSOR_TRIG_DATA_READY;
+	trig.chan = SENSOR_CHAN_ACCEL_XYZ;
+
+	if (sensor_trigger_set(itds, &trig, trigger_handler) < 0) {
+		printf("cannot set trigger.\n");
+		return;
+	}
+
+	remaining_test_time = MAX_TEST_TIME;
+	do {
+		k_sleep(K_MSEC(SLEEPTIME));
+		remaining_test_time -= SLEEPTIME;
+	} while (remaining_test_time > 0);
+
+	trig.type = SENSOR_TRIG_DATA_READY;
+	trig.chan = SENSOR_CHAN_ACCEL_XYZ;
+	if (sensor_trigger_set(itds, &trig, NULL) < 0) {
+		printf("cannot clear trigger.\n");
+		return;
+	}
+
+	printf("Data ready trigger test finished.\n");
+#endif
+
+}
+
+void main(void)
+{
+	struct device *itds;
+	struct sensor_value attr;
+
+	printf("get device wsen-itds\n");
+	itds = device_get_binding("WSEN-ITDS");
```

Comment:
Use the device tree label macro instead of hardcoding the string. e.g., 
```suggestion
	itds = device_get_binding(DT_LABEL(DT_INST(0, we_wsen_itds)));
```

Reply:
This still needs to be done

Reply:
I missed it, now addressed.